### PR TITLE
Show whether a PR is WIP inside popups 

### PR DESCRIPTION
**Pull request by bramhaag**
Fixes https://codeberg.org/forgejo/forgejo/issues/2257

Draft status of a PR is currently not exposed by the API. This PR adds a 'draft' field to pull requests in the API, which is used to correctly set the PR color/icon in a ContextPopup.

---

Before:
image

After:
image

### CODE REVIEW

File: modules/structs/issue.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ const (
 type PullRequestMeta struct {
 	HasMerged bool       `json:"merged"`
 	Merged    *time.Time `json:"merged_at"`
+	IsDraft   bool       `json:"draft"`
```
Comment (lunny) [inline modules/structs/issue.go]: I think `IsWIP` is a better name.

Comment (bramhaag) [reply modules/structs/issue.go]: Renamed `IsDraft` to `IsWorkInProgress` for consistency with [PullRequest.IsWorkInProgress](https://github.com/go-gitea/gitea/blob/2a50d780f93957223da9494c61bb3181ffa7ff65/models/issues/pull.go#L689)

Renamed json field `draft` to `wip` for consistency with the terminology used in the docs.

**Review by lunny — APPROVED** [review]

File: modules/structs/issue.go [hunk 2]
```diff
@@ -26,8 +26,9 @@ const (
 
 // PullRequestMeta PR info if an issue is a PR
 type PullRequestMeta struct {
-	HasMerged bool       `json:"merged"`
-	Merged    *time.Time `json:"merged_at"`
+	HasMerged        bool       `json:"merged"`
+	Merged           *time.Time `json:"merged_at"`
+	IsWorkInProgress bool       `json:"wip"`
```
Comment (delvh) [inline modules/structs/issue.go]: I am against this naming as we intend to be as GitHub API-compatible as possible, unless there is a good reason to deviate.
In the GitHub API, this field is called `draft`

**Review by delvh — CHANGES_REQUESTED** [review]

File: modules/structs/issue.go (hunk 2 above)
Comment (lunny) [reply modules/structs/issue.go]: Can you post the API URL?

Comment (jolheiser) [reply modules/structs/issue.go]: afaik we only use `Draft` for releases.
Generally we refer to it as `WIP` in other code/docs for PRs.

-----

That being said, GitHub does refer to this (similar) field as `draft`. I'm fine with whichever, though I think personally I'd rather be self-consistent than GitHub-consistent in the event of a field collision like this.

Comment (delvh) [reply modules/structs/issue.go]: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests
Navigate to `response schema` and search for `draft`

Comment (bramhaag) [reply modules/structs/issue.go]: @delvh did @jolheiser's edit change your mind? 

If not, I propose I rename `wip` to `draft` to stay consistent with GitHub's naming, and leave `IsWorkInProgress` as is to stay consistent with the rest of the codebase.

Comment (delvh) [reply modules/structs/issue.go]: Hmm… Difficult question:
Do we want to be GitHub API compliant, or do we want to follow our own terminology?
I think it is better if we stay compliant.
Especially now that we have Actions, every deviation will be noticeable in actions that are written to talk with GitHubs API…

Comment (lunny) [reply modules/structs/issue.go]: Or maybe we can have both fields and check wip first and then draft.

Comment (bramhaag) [reply modules/structs/issue.go]: I'm not sure if I understand the second part. Are `wip` and `draft` not synonymous in this case? Why would you check both?

Comment (jolheiser) [reply modules/structs/issue.go]: I'm fine with it remaining compatible. I think having duplicate fields would just be overcomplicated, so `draft` works for me. :+1:

**Review by KN4CK3R — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**